### PR TITLE
Implement `forward_volume` and `reverse_volume` properties in Surface

### PR DESCRIPTION
**Pull request by paulromano**
This PR adds `forward_volume` and `reverse_volume` properties in the `Surface` class, which in turn rely on a lower-level `surf_sense` property. There was also a small bug in the `get_volumes()` method that is now fixed.

### CODE REVIEW

**Comment by pshriwise** [comment]
Looking at the original `Surface.get_volumes` method again, using a `get_parent_meshsets` call feels a little silly to me when the data is also stored as part of the sense tag. Maybe that's more to do with the file spec really as the information exists in both places, but only the sense tag maintains the ordering needed to determine which volume has a forward or reverse sense wrt the surface.

Just throwing this out there, what if we replace the `Surface.get_volumes` method with something like `Surface.parent_volumes` that always returns a list of two `Volume` classes with the forward/reverse ordering by querying the sense tag data rather than the meshset parent-child relationships? That way the sense information is always provided. The `forward_volume` and `reverse_volume` properties are still good IMO as they're more explicit, so we could keep those too.

File: dagmc/dagnav.py
Comment (pshriwise) [inline dagmc/dagnav.py:304]: Is this currently the only public method that returns MOAB `EntityHandles`? I think it might be better to stick with the `Volume` objects or make this an internal method to stay consistent.

File: dagmc/dagnav.py
Comment (pshriwise) [inline dagmc/dagnav.py:313]: When we establish this relationship with the new volumes via the sense tag, we should also update the surface's parent-child relationships as well.

**Review by pshriwise — CHANGES_REQUESTED** [review]
A couple of thoughts here, but generally looks good! Thanks for catching that `get_volumes` bug.

File: dagmc/dagnav.py
Comment (paulromano) [reply dagmc/dagnav.py:304]: Good idea -- I'll update this to use `Volume` instead of handles directly, which also has the benefit of consolidating the checks for 0 handles in one place.

File: dagmc/dagnav.py
Comment (paulromano) [reply dagmc/dagnav.py:318]: Just updated this to add the parent-child relationships -- I think I did this correctly but would appreciate you double-checking

**Comment by paulromano** [comment]
@pshriwise Comments have been addressed -- let me know if anything further is needed

File: dagmc/dagnav.py
Comment (pshriwise) [inline dagmc/dagnav.py:330]: What's the scenario in which `surf_sense` would raise a `RuntimeError`?

Comment (paulromano) [reply dagmc/dagnav.py:330]: I think I ran into this before the GEOM_SENSE_2 tag had been set at all for a given surface. Looking at it now, I feel like this check could be part of the `surf_sense` getter though so let me see if I can fix that up.

Comment (paulromano) [reply dagmc/dagnav.py]: Ok, I moved the `try/except` block into the getter for `surf_sense`, which looks a lot better!

File: test/test_basic.py
Comment (gonuke) [inline test/test_basic.py:129]: Should we also test the setters?

**Review by gonuke — CHANGES_REQUESTED** [review]
This all looks ok to me, but suggest a little more testing

File: test/test_basic.py
Comment (paulromano) [reply test/test_basic.py:129]: Good call -- I'll add a check for the setters

**Review by pshriwise — APPROVED** [review]
Thanks @paulromano! I'll let @gonuke approve and merge when/if ready.

**Review by gonuke — APPROVED** [review]
Thanks for the additional tests @paulromano

**Pull request merged**
